### ISSUE TITLE
Backend redesign

### ISSUE THREAD

**Issue by athairus** [team-phoenix/Phoenix#162]
To keep the QML thread from being blocked by `Core`, we need to move it and its controller, `VideoItem`, to a different thread. Unfortunately, that can't be done as QML objects must live in the QML thread. An alternative approach is to split `VideoItem` into separate classes: an object that lives in the QML thread and subclasses QQuickItem, `GameManager`, and an object that lives in the core thread and controls `Core`'s execution, `GameSession`.

`Core` will be renamed to `LibretroCore`.

`GameManager` interacts with other QML objects via a set of properties and functions. Internally, it'll control the lifecycle of `GameSession` act as a proxy of sorts passing messages along to it.

`GameSession` interacts with `GameManager` via a signal-slot interface to keep things thread-safe. It controls the lifecycle of `LibretroCore`. `GameSession` and `LibretroCore` communicate via function calls, as they will always live in the same thread.

**Comment by athairus** [comment]
I'm also going to rename `VideoItem` to `VideoOutput`

**Comment by athairus** [comment]
I think it'll make things simpler if we make the QML interface as simple as possible. In the future, games shouldn't be launched from QML by setting `core` and `game` path properties but instead a launch should be triggered by setting the current game UUID to a new value, but only once `CoreControl` is `CoreControl.STOPPED`.

**Comment by athairus** [comment]
`CoreControl` MUST be started from the C++ side, otherwise it's locked (?) into the QML thread.

**Comment by athairus** [comment]
Okay, let's not use a UUID, but instead just leave the door open for future backend implementations by providing a `type` string to the tuple passed to `CoreController`.

**Comment by athairus** [comment]
Also, `GameManager` should not handle video output. That job should lie in another class's job, like `VideoOutput`.

**Comment by athairus** [comment]
Putting this into a new issue: #207
